velox/functions/prestosql/window/LeadLag.cpp

```diff
@@ -141,25 +141,41 @@ class LeadLagFunction : public exec::WindowFunction {
       if (offsets_->isNullAt(i)) {
         rowNumbers_[i] = kNullRow;
       } else {
-        vector_size_t offset = offsets_->valueAt(i);
+        auto offset = offsets_->valueAt(i);
         VELOX_USER_CHECK_GE(offset, 0, "Offset must be at least 0");
 
         if constexpr (isLag) {
           if constexpr (ignoreNulls) {
-            rowNumbers_[i] = rowNumberIgnoreNull(
-                rawNulls, offset, partitionOffset_ + i - 1, -1, -1);
+            // Calculate the maximum possible row number to see if the offset
```

Comment:
We can just simply do at line 144 the following code. It should solve the problem at hand.

```
auto offset = offsets_->valueAt(i);
if (offset > partition_->numRows) {
  rowNumbers_[i] = kNullRow;
  return;
}
```

Reply:
Yeah, your solution is simpler, updated the PR.